adapters/appnexus/appnexus.go

```diff
@@ -260,17 +263,11 @@ func preprocess(imp *openrtb2.Imp, appnexusExt *openrtb_ext.ExtImpAppnexus, defa
 	}
 
 	// Populate imp.displaymanagerver if the SDK failed to do it.
-	if len(imp.DisplayManagerVer) == 0 && len(defaultDisplayManagerVer) > 0 {
-		imp.DisplayManagerVer = defaultDisplayManagerVer
+	if len(imp.DisplayManagerVer) == 0 && len(displayManagerVer) > 0 {
+		imp.DisplayManagerVer = displayManagerVer
 	}
 
-	impExt := appnexusImpExt{Appnexus: appnexusImpExtAppnexus{
-		PlacementID:       int(appnexusExt.PlacementId),
-		TrafficSourceCode: appnexusExt.TrafficSourceCode,
-		Keywords:          makeKeywordStr(appnexusExt.Keywords),
-		UsePmtRule:        appnexusExt.UsePaymentRule,
-		PrivateSizes:      appnexusExt.PrivateSizes,
-	}}
+	impExt := buildImpExt(appnexusExt)
```

Comment:
Here `buildImpExt` only initialises appnexusImpExt. This can happen in buildRequestImp. No need for separate function

Reply:
There is already a lot happening in buildRequestImp. Setting Impression extension in a different function sounds ok to me.

Reply:
fixed here - https://github.com/prebid/prebid-server/pull/2668/commits/54b627ec9166b55039ca6a0379cb9cf7e60c93b4